script.js

```diff
@@ -1,4 +1,117 @@
+//input -- text of 'scissors' 'paper' 'stone'
+// if input = random == its a draw
+// if input is not within the 3 texts, let them know its wrong & provide the options
+// system generates random option
+//output -- wins if SPS logic makes sense + tells user if they won/lost/draw
+
+// build code to generate random number selection, number would be from 1 to 3, 3 inclusive
+var generateSystemNumber = function () {
+  var systemNumber = Math.floor(Math.random() * 3) + 1;
+  console.log("random number generated" + systemNumber); // random number will be generated from 1 to 3
+  return systemNumber;
+};
+
+// build code for the 3 options
+var generateSystemChoice = function () {
+  var choiceCount = 3;
+  var systemChoice = generateSystemNumber(choiceCount);
+  console.log("random number chosen" + systemChoice); //this will generate random number based on the no. of choices
+  if (systemChoice == 1) {
+    return "Scissors ✂️";
+  }
+  if (systemChoice == 2) {
+    return "Paper 🧻";
+  }
+  if (systemChoice == 3) {
+    return "Stone 💎";
+  }
+  return systemChoice;
+};
+
+// adding state to track no of times user wins
+var winTracker = 0;
+var numberOfTries = 0;
+var userName = "";
+var currentGameMode = "waiting for player";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = generateOutcome(input);
+  console.log("generate outcome - username");
+
+  // To set code for Username input first before Dice game starts
+  if (currentGameMode == "waiting for player") {
+    userName = input;
+    currentGameMode = "game start";
+    myOutputValue =
+      "Welcome " +
+      userName +
+      ", the game starts now! <br><br> Please input Scissors ✂️, Paper 🧻 or Stone 💎 to play.";
+  }
```

Comment:
Great that the main function is clean and short.  Do remember that it is a good practice to declare the functions prior to using them.  

Because main function is using the generateOutcome function. Main function should be moved below. This is mainly for code readability